.github/workflows/build-macos.yml

```diff
@@ -0,0 +1,81 @@
+---
+# CI code for build and test on macOS
+name: macOS Build and test
+on:
+  push: # Master branch checks only validate the build and generate artifacts for testing.
+    branches:
+      - master
+  pull_request: null # PR checks only validate the build and generate artifacts for testing.
+  workflow_dispatch: # Dispatch runs build and validate, then push to the appropriate storage location.
+    inputs:          # Inputs for this workflow can be omitted
+      type:
+        description: Build Type
+        default: nightly
+        required: true
+      version:
+        description: Version Tag
+        default: nightly
+        required: true
+concurrency: 
+  group: ${{ github.workflow }}-${{ github.ref }}
+jobs:
+  build-test:
+    runs-on: ${{ matrix.runner }}
+    strategy:
+      fail-fast: false
+      max-parallel: 3
+      matrix:
+        include:
+          - name: macos-12
+            runner: macos-12
+          - name: macos-13
+            runner: macos-13
+          - name: macos-14-M1
+            runner: macos-14
+    steps:
+      - uses: actions/checkout@v4
+        id: checkout
+        with:
+          submodules: recursive
+      - name: Setup fundamental depedencies
+        id: setup-runner
+        run: |
+          brew install bash
+      - name: Install netdata dependencies
+        id: install-nd-dep
+        run: |
+          export DISABLE_TELEMETRY=1
+          export GITHUB_ACTIONS=true
+          bash ./packaging/installer/install-required-packages.sh --dont-wait --non-interactive netdata
+      - name: Build from source
+        id: build-source
+        run: |
+          sudo bash ./netdata-installer.sh --install-no-prefix /usr/local/netdata --dont-wait --dont-start-it --require-cloud --one-time-build
+      - name: Test Agent start up
+        id: test-agent
+        run: |
+          /usr/local/netdata/usr/sbin/netdata -D > ./netdata.log 2>&1 &
+          ./packaging/runtime-check.sh || exit 1
```

Comment:
The runtime check script should be updated to handle this properly instead of starting the agent outside of it. Also, unless I’m mistaken, you should not need the `|| exit 1` here.

Reply:
We use this script in different places, leave it like that for now because the prefixes may be tricky